src/content/app/species-selector/views/species-manager/selected-genomes-table/SelectedGenomesTable.tsx

```diff
@@ -14,10 +14,346 @@
+                  className={styles.toggle}
+                  isOn={species.isEnabled}
+                  onChange={() => toggleGenomeUse(species)}
+                  disabled={isInDeletionMode}
+                />
+              </td>
+            </tr>
+            {isInDeletionMode &&
+              tableState.deletionModeSettings?.rowDeletionInitiated ===
+                species.genome_id && (
+                <ConfirmDeletion
+                  species={species}
+                  tableColumnsCount={8}
+                  onDelete={deleteGenomes}
+                  onCancel={exitDeletionMode}
+                />
+              )}
+          </Fragment>
+        ))}
+      </tbody>
+    </Table>
+  );
 };
 
+const DeleteButtonOrCheckbox = ({
+  species,
+  isInDeletionMode,
+  isMarkedForDeletion,
+  enterDeletionMode,
+  addGenomeToDeleteList,
+  removeGenomeFromDeleteList
+}: {
+  species: CommittedItem;
+  isInDeletionMode: boolean;
+  isMarkedForDeletion?: boolean;
+  enterDeletionMode: (species: CommittedItem) => void;
+  addGenomeToDeleteList: (species: CommittedItem) => void;
+  removeGenomeFromDeleteList: (species: CommittedItem) => void;
+}) => {
+  const onCheckboxChange = (isChecked: boolean) => {
+    if (isChecked) {
+      addGenomeToDeleteList(species);
+    } else {
+      removeGenomeFromDeleteList(species);
+    }
+  };
+
+  if (!isInDeletionMode) {
+    return <DeleteButton onClick={() => enterDeletionMode(species)} />;
+  } else {
+    return (
+      <Checkbox checked={!!isMarkedForDeletion} onChange={onCheckboxChange} />
+    );
+  }
+};
+
+/**
+ * This is visually represented as a table row, which always has two columns
+ * (the start of the second column aligns with the second to last column of the table)
+ */
+const ConfirmDeletion = (props: {
+  species: CommittedItem;
+  tableColumnsCount: number; // this will be different depending on whether showing all columns or is hiding some
+  onDelete: (species: CommittedItem) => void;
+  onCancel: () => void;
+}) => {
+  // this row will use the two rightmost columns of the table,
+  // but will merge all the columns to the left
+  const cpanColumnsCount = props.tableColumnsCount - 2;
+
+  const onDelete = () => {
+    props.onDelete(props.species);
+  };
+
+  // FIXME: the colspan on the first td should depend on how many columns the table shows
+  return (
+    <tr className={styles.removalRow}>
+      <td colSpan={cpanColumnsCount}>
+        <div className={styles.removalRowContent}>
+          <span className={styles.removalWarning}>{removalWarning}</span>
+          <PrimaryButton onClick={onDelete}>Remove</PrimaryButton>
+        </div>
+      </td>
+      <td colSpan={2}>
+        <button className={styles.cancelGenomeRemoval} onClick={props.onCancel}>
+          Do not remove
+        </button>
+      </td>
+    </tr>
+  );
+};
+
+const removalWarning =
+  'Any configuration of views for this species will be lost if you remove it - do you wish to continue?';
```

Comment:
Any or All?

Reply:
🤷  The [XD](https://xd.adobe.com/view/2ab31a29-bcf3-4c8f-b50c-906a3949cfe2-7fe0/screen/2b49466b-a058-4484-8aea-0da2b01db7ca) says any. It's a question for Andrea, really.